### PR TITLE
Fix opinionated usage of logging module

### PR DESCRIPTION
**Pull request by Nicholas-Mitchell**
1. Remove all usages of `logging.basicConfig` that are outside of the `__main__` block of python files.
2. Add `logging.basicConfig` inside each `__main__` block of python files.
3. Remove usage and imports of the `LogRecord` class.

Please see the related issue for all details:  https://github.com/isl-org/Open3D-ML/issues/330

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/418)

### CODE REVIEW

**Comment by sanskar107** [comment]
@Nicholas-Mitchell thanks for your contribution. Could you update your PR to fix style check?
You can run `make apply-style` to fix these errors.

**Comment by Nicholas-Mitchell** [comment]
I saw the errors and it doesn't actually say anything useful e.g. What is actually violating style expectations.

Have you got some more detailed instructions on how to run that cleanup tool please?

**Comment by sanskar107** [comment]
You can run `make apply-style` in the repository root, It will apply all the style fixes. Also if you want to see the proposed changes, you can run `yapf -d *.py --recursive`.

File: setup.py [hunk 1]
```diff
@@ -11,6 +12,12 @@
 assert torch_ver >= [1, 4], "Requires PyTorch >= 1.4"
 
 if __name__ == '__main__':
+
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(levelname)s - %(asctime)s - %(module)s - %(message)s',
+    )
+
```
Comment (sanskar107) [inline setup.py:20]: Do we need this in `setup.py` ?

**Review by sanskar107 — APPROVED** [review]

**Comment by Nicholas-Mitchell** [comment]
Apologies - haven't had time to get back to this PR.

logging.BasicConfig  can be used in any main section. Perhaps not needed in a setup file, but also doesn't hurt.

**Comment by Nicholas-Mitchell** [comment]
@sanskar107 @benjaminum - ping on this - does everything look ok now?

**Pull request merged**